### DISCUSSION TITLE
Why autopipeline and text2img pipeline cost time much different?

### DISCUSSION THREAD

**Discussion by zengjie617789** [huggingface/diffusers#7826] [Q&A]
I used the autopipeline and text2imgpipeline to load model respectively, but the time is much different. Here are code I used and the time screenshot：
```
pipe = StableDiffusionXLPipeline.from_pretrained(data_dir,  use_safetensors=True, variant="fp16").to("cuda")
pipe= AutoPipelineForText2Image.from_pretrained(data_dir, torch_dtype=torch.float16, variant="fp16", use_safetensors=True).to("cuda")

```
![image](https://github.com/huggingface/diffusers/assets/25050291/2624cbc1-9aab-44dd-9f42-3c9ce94e2a16)

![image](https://github.com/huggingface/diffusers/assets/25050291/fcb948c0-e501-44e4-9e19-4807426c594d)

the same prompt and steps.
Any help will be appreciated.

**Comment by tolgacangoz** [comment]
To equalize them totally, one needs to add `torch_dtype=torch.float16` to the first `pipe`; because the default is `torch_dtype=torch.float32`.
